src/cast_core/include/cast_core/actors/MonotonicLoader.hpp

```diff
@@ -33,7 +33,20 @@ namespace genny::actor {
  * indexes. Each collection is identically configured. The document shape, number of documents,
  * number of collections, and list of indexes are all adjustable from the yaml configuration.
  *
- * This actor is identical to `Loader` except it loads with monotonically increasing _ids.
+ * This actor is identical to `Loader` except it loads with monotonically increasing _ids and it has
+ * an additional pair of arguments ('FieldIncreasingByOffsetFromID' and 'OffsetFromID') to generate
+ * a field in each document whose value is 'OffsetFromID' greater than the '_id' of the document.
+ * For example, if the field name is 'a' and the offset is 1, the documents will look like
+ *      {
+ *          "_id": 1,
+ *          "a": 2,
+ *          ...
+ *      },
+ *      {
+ *          "_id": 2,
+ *          "a": 3,
+ *          ...
+ *      }, ...
```

Comment:
The idea behind this change was to allow easy calculation of result set size for the $graphLookup queries with `maxDepth > 0`, see my comment below for more details. But I _think_ this has had the unfortunate side effect of reducing the difference in performance for the $graphLookups with `maxDepth = 0` vs `maxDepth > 0`.

I have not confirmed this, but if the cache maintained by DocumentSourceGraphLookup looks like a mapping from `valueOfConnectToField` to `documentInForeignCollection`, then after just 1 iteration, all of the documents in the foreign collection should be in the cache and we can do iterations without having to do any additional querying. If this is true, then there seems to be a trade off between meaningfully testing `maxDepth > 0` and correctly bounding our result set size.

Reply:
Hmm yeah that would explain the benchmark results that we're seeing for recursive vs non-recursive $graphLookup. Perhaps instead of using the MonotonicLoader we could instead construct the collection such that the first iteration of the search only surfaces a portion of the total documents so that all of them aren't added to the cache at depth 0 - for example, if we construct documents with `{a: {^RandomInt: {min: 1, max: 100}}, b: {^RandomInt: {min: 1, max: 100}}, c: {^RandomInt: {min: 1, max: 10}}}` and we join on `a` and `b` but start with `c`. With this approach it might not be possible to completely avoid filling the cache such that later iterations wouldn't mostly consult the cache instead of doing additional querying, but I'm okay with that. This would at least give us a predictable measure of how the result set would grow so that we can continue bounding the result size of the recursive $graphLookup workloads using the `maxDepth` field.

Another option would be to add a new Loader Actor that would give us some control over what fraction of the generated documents have certain value ranges for each field. This way we could artificially ensure that new documents are introduced to the frontier for each iteration of the search. I haven't thought through all the details but this approach would obviously require more changes than the one above. Personally I'm not sure if this is worth the effort, but of either of you feel differently I could be convinced!